src/hardware/openbench-logic-sniffer/protocol.c

```diff
@@ -556,17 +560,22 @@ SR_PRIV int ols_receive_data(int fd, int revents, void *cb_data)
 		}
 
 		/* Send post-trigger / all captured samples. */
-		int num_pre_trigger_samples = devc->trigger_at_smpl ==
-							      OLS_NO_TRIGGER ?
-							    0 :
-							    devc->trigger_at_smpl;
-		packet.type = SR_DF_LOGIC;
-		packet.payload = &logic;
-		logic.length =
-			(devc->num_samples - num_pre_trigger_samples) * 4;
-		logic.unitsize = 4;
-		logic.data = devc->raw_sample_buf + num_pre_trigger_samples * 4;
-		sr_session_send(sdi, &packet);
+		num_pre_trigger_samples =
+			devc->trigger_at_smpl == OLS_NO_TRIGGER ?
+				      0 :
+				      MIN((unsigned int)devc->trigger_at_smpl,
+				    devc->num_samples);
+		if (devc->num_samples > num_pre_trigger_samples) {
+			packet.type = SR_DF_LOGIC;
+			packet.payload = &logic;
+			logic.length =
+				(devc->num_samples - num_pre_trigger_samples) *
+				4;
+			logic.unitsize = 4;
+			logic.data = devc->raw_sample_buf +
+				     num_pre_trigger_samples * 4;
+			sr_session_send(sdi, &packet);
+		}
```

Comment:
Looks good. Maybe commit message could state what non-desired behaviour happened with the old code?

Reply:
I honestly don't remember what happened when we send garbage via sr_session_send (trigger in the future). I guess it is some unrecoverable state, like a crash, otherwise I think I wouldn't have fixed it.